### ISSUE TITLE
error/ loss aways equal 87.3362. I change the lr but not work. 

### ISSUE THREAD

**Issue by mqchen1993** [razorx89/caffe-wrn-generator#1]
I1128 08:59:24.073550 14972 solver.cpp:397]     Test net output #0: Accuracy1 = 1
I1128 08:59:24.073571 14972 solver.cpp:397]     Test net output #1: SoftmaxWithLoss1 = 87.3362 (* 1 = 87.3362 loss)
I1128 08:59:24.187857 14972 solver.cpp:218] Iteration 1170 (6.24905 iter/s, 62.4095s/390 iters), loss = 87.3363
I1128 08:59:24.187876 14972 solver.cpp:237]     Train net output #0: SoftmaxWithLoss1 = 87.3365 (* 1 = 87.3365 loss)
I1128 08:59:24.187881 14972 sgd_solver.cpp:105] Iteration 1170, lr = 0.1
I1128 09:00:11.118723 14972 solver.cpp:330] Iteration 1560, Testing net (#0)
I1128 09:00:26.383985 14979 data_layer.cpp:73] Restarting data prefetching from start.
I1128 09:00:26.506765 14972 solver.cpp:397]     Test net output #0: Accuracy1 = 1
I1128 09:00:26.506784 14972 solver.cpp:397]     Test net output #1: SoftmaxWithLoss1 = 87.3362 (* 1 = 87.3362 loss)
I1128 09:00:26.621100 14972 solver.cpp:218] Iteration 1560 (6.24695 iter/s, 62.4305s/390 iters), loss = 87.3363
I1128 09:00:26.621127 14972 solver.cpp:237]     Train net output #0: SoftmaxWithLoss1 = 87.3365 (* 1 = 87.3365 loss)
I1128 09:00:26.621134 14972 sgd_solver.cpp:105] Iteration 1560, lr = 0.1
I1128 09:01:13.576076 14972 solver.cpp:330] Iteration 1950, Testing net (#0)
I1128 09:01:33.771523 14979 data_layer.cpp:73] Restarting data prefetching from start.
I1128 09:01:34.000592 14972 solver.cpp:397]     Test net output #0: Accuracy1 = 1
I1128 09:01:34.000615 14972 solver.cpp:397]     Test net output #1: SoftmaxWithLoss1 = 87.3362 (* 1 = 87.3362 loss)
I1128 09:01:34.220914 14972 solver.cpp:218] Iteration 1950 (5.76949 iter/s, 67.597s/390 iters), loss = 87.3363
I1128 09:01:34.220949 14972 solver.cpp:237]     Train net output #0: SoftmaxWithLoss1 = 87.3365 (* 1 = 87.3365 loss)
I1128 09:01:34.220955 14972 sgd_solver.cpp:105] Iteration 1950, lr = 0.1
I1128 09:03:16.823488 14972 solver.cpp:330] Iteration 2340, Testing net (#0)
I1128 09:03:48.873394 14979 data_layer.cpp:73] Restarting data prefetching from start.
I1128 09:03:49.115883 14972 solver.cpp:397]     Test net output #0: Accuracy1 = 1
I1128 09:03:49.115906 14972 solver.cpp:397]     Test net output #1: SoftmaxWithLoss1 = 87.3362 (* 1 = 87.3362 loss)
I1128 09:03:49.344960 14972 solver.cpp:218] Iteration 2340 (2.88635 iter/s, 135.119s/390 iters), loss = 87.3363

**Comment by razorx89** [comment]
Without further information about your training setup, I am unable to help you. What does your solver.prototxt look like? Which network did you use? On which dataset are you training? A constant high loss value normally indicates an issue with the optimizer settings (too high learning rate or weight decay).

**Comment by mqchen1993** [comment]
I trained on the cifar10.
Attached is my prototxt file.

Thanks!

2017-11-28 17:33 GMT+08:00 Sven Koitka <notifications@github.com>:

**Comment by razorx89** [comment]
No, it is not.

**Comment by mqchen1993** [comment]
Hello, why not this, this can not be used in cifar10.
I downloaded from your homepage.

Thanks!



​

2017-11-28 17:55 GMT+08:00 Sven Koitka <notifications@github.com>:

**Comment by razorx89** [comment]
You did not attach any prototxt file above. Furthermore, I requested the solver.prototxt and not trainval.prototxt (solver.prototxt is not included in the examples).

**Comment by mqchen1993** [comment]
I am sorry, it may be my side of the transmission error. I'll pass it again.

Thanks！



net: "examples/cifar10/wrn/cifar100_WRN-16-4_dropout_train_val.prototxt"
test_iter: 100
test_interval: 1000
base_lr: 0.0001
#clip_gradients:30
display: 100
max_iter: 64000
lr_policy: "multistep"
gamma: 0.1
momentum: 0.9
weight_decay: 0.0005
snapshot: 10000
snapshot_prefix: "examples/cifar10/wrn"
solver_mode: GPU
test_initialization: true
stepvalue: 32000
stepvalue: 48000

2017-11-28 18:53 GMT+08:00 Sven Koitka <notifications@github.com>:

**Comment by razorx89** [comment]
Your learning rate seems very low. I had a look at my old experiments and here is one of my solver configs:
```
net: "cifar100_WRN-16-8_train_val.prototxt"
max_iter: 46875

test_iter: 100
test_interval: 391
test_initialization: false

display: 10
average_loss: 10

solver_mode: GPU
type: "Nesterov"
momentum: 0.9
base_lr: 0.1
weight_decay: 0.00025
lr_policy: "step"
stepsize: 15625
gamma: 0.1

snapshot: 3907
snapshot_prefix: "snapshot_cifar100_WRN-16-8"
```
Since you are specifying momentum you should also specify the solver type (defaults to SGD https://github.com/BVLC/caffe/blob/master/src/caffe/proto/caffe.proto#L212). Besides that it could be that your input data has errors. Please check on how you are building your lmdbs.

**Comment by mqchen1993** [comment]
I modified the solver file as you gave it. But still failed. Dataset is no
problem, because I use other network training is possible.

Thanks！

2017-11-28 19:08 GMT+08:00 Sven Koitka <notifications@github.com>:
